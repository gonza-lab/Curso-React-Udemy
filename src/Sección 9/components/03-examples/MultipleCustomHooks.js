import React from 'react';
import { useCounter } from '../../hooks/useCounter';
import { useFetch } from '../../hooks/useFetch';

export const MultipleCustomHooks = ({ id = 1 }) => {
  const [counter, incr] = useCounter(1);

  const { data, isLoading } = useFetch(
    `https://www.breakingbadapi.com/api/quotes/${counter}`
  );

  const { author, quote } = !!data && data[0];

  return (
    <div>
      <h1>Breaking Bad Quotes</h1>
      <hr />

      {isLoading ? (
        <div className="alert alert-info text-center">Loading...</div>
      ) : (
        <blockquote className="blockquote text-right">
          <p className="mb-0">{quote}</p>
          <footer className="blockquote-footer">{author}</footer>
        </blockquote>
      )}

      <button className="btn btn-success" onClick={incr}>Siguiente frase</button>
    </div>
  );
};
