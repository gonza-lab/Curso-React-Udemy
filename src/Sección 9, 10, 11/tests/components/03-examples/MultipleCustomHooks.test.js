import { shallow } from 'enzyme';
import React from 'react';
import { MultipleCustomHooks } from '../../../components/03-examples/MultipleCustomHooks';
import { useFetch } from '../../../hooks/useFetch';
jest.mock('../../../hooks/useFetch');

describe('Pruebas en MultipleCustomHooks', () => {
  test('Debe mostrar el componente correctamente', () => {
    useFetch.mockReturnValue({
      data: null,
      isLoading: true,
      error: null,
    });

    expect(shallow(<MultipleCustomHooks />)).toMatchSnapshot();
  });

  test('Debe mostrar la informacion correctamente', () => {
    useFetch.mockReturnValue({
      data: [
        {
          quote: 'Si estas triste, deja de estar triste',
          author: 'Gonzalo Flores',
        },
      ],
      isLoading: false,
      error: null,
    });
    expect(shallow(<MultipleCustomHooks />)).toMatchSnapshot();
  });
});
