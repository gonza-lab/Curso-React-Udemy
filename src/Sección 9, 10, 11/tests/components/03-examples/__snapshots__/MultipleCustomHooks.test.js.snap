// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en MultipleCustomHooks Debe mostrar el componente correctamente 1`] = `
<div>
  <h1>
    Breaking Bad Quotes
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loading...
  </div>
  <button
    className="btn btn-success"
    onClick={[Function]}
  >
    Siguiente frase
  </button>
</div>
`;

exports[`Pruebas en MultipleCustomHooks Debe mostrar la informacion correctamente 1`] = `
<div>
  <h1>
    Breaking Bad Quotes
  </h1>
  <hr />
  <blockquote
    className="blockquote text-right"
  >
    <p
      className="mb-0"
    >
      Si estas triste, deja de estar triste
    </p>
    <footer
      className="blockquote-footer"
    >
      Gonzalo Flores
    </footer>
  </blockquote>
  <button
    className="btn btn-success"
    onClick={[Function]}
  >
    Siguiente frase
  </button>
</div>
`;
